hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -297,6 +297,10 @@ final void initInvocationTarget() throws Exception {
         }
     }
 
+    private boolean invocationShouldHaveTargetMember() {
```

Comment:
Method name seems misleading to me because there is no invocation without a target, every invocation must have a target, we only want to check here whether the target is required to be a member of this cluster at the time of the check.

So an alternative naming can be `requiresTargetAsClusterMember` or `requiresTargetAsMemberOfThisCluster` or some better name that you find.

Reply:
https://github.com/hazelcast/hazelcast/pull/23005/commits/29667c5d77381f80aae74a1b9c11666627c033d4